pkg/pillar/volumehandlers/csihandler.go

```diff
@@ -224,31 +224,64 @@ func (handler *volumeHandlerCSI) CreateVolume() (string, error) {
 func (handler *volumeHandlerCSI) DestroyVolume() (string, error) {
 	pvcName := handler.status.GetPVCName()
 	handler.log.Noticef("DestroyVolume called for PVC %s", pvcName)
-	err := kubeapi.DeletePVC(pvcName, handler.log)
-	if err != nil {
-		// Its OK if not found since PVC might have been deleted already
-		if kerr.IsNotFound(err) {
-			handler.log.Noticef("PVC %s not found, might have been deleted", pvcName)
-			return pvcName, nil
-		} else {
-			return pvcName, err
+	// if this is a replicated volume, do not delete PVC on this node.
+	if !handler.status.IsReplicated {
+		err := kubeapi.DeletePVC(pvcName, handler.log)
+		if err != nil {
+			// Its OK if not found since PVC might have been deleted already
+			if kerr.IsNotFound(err) {
+				handler.log.Noticef("PVC %s not found, might have been deleted", pvcName)
+				return pvcName, nil
+			} else {
+				return pvcName, err
+			}
 		}
+	} else { // for debug
+		handler.log.Noticef("DestroyVolume container skip delete PVC %s", pvcName)
```

Comment:
if this message it's only for debug, then you can just use `log.Debugf` and remove the comment....